hydrus/tests/test_crud_general.py

```diff
@@ -0,0 +1,243 @@
+"""Unit tests for CRUD operations in hydrus.data.crud."""
+
+import unittest
+
+from sqlalchemy import create_engine
+from sqlalchemy.orm import sessionmaker, scoped_session
+import hydrus.data.crud as crud
+from hydrus.data.db_models import Base
+from hydrus.data import doc_parse
+from hydrus.hydraspec import doc_maker
+from hydrus.samples.hydra_doc_sample import doc as api
+import random
+from typing import List
+import string
+
+
+def gen_dummy_object(class_, doc):
+    """Create a dummy object based on the definitions in the API Doc."""
+    object_ = {
+        "@type": class_
+    }
+    if class_ in doc.parsed_classes:
+        for prop in doc.parsed_classes[class_]["class"].supportedProperty:
+            if "vocab:" in prop.prop:
+                prop_class = prop.prop.replace("vocab:", "")
+                object_[prop.title] = gen_dummy_object(prop_class, doc)
+            else:
+                object_[prop.title] = ''.join(random.choice(
+                    string.ascii_uppercase + string.digits) for _ in range(6))
+        return object_
+
+
+class TestCRUD(unittest.TestCase):
+    """Test class for CRUD Tests."""
+
+    @classmethod
+    def setUpClass(self):
+        """Database setup before the CRUD tests."""
+        print("Creating a temporary datatbsse...")
+        engine = create_engine('sqlite:///:memory:')
+        Base.metadata.create_all(engine)
+        session = scoped_session(sessionmaker(bind=engine))
+        id_list = api["@id"].split('/')
+        id_list.pop()
+        id_list.pop()
+        host = '/'.join(id_list)[0:]
+        self.HYDRUS_SERVER_URL = host+'/'
+        self.API_NAME = api["@id"].split('/')[len(api["@id"].split('/')) - 2] + '/'
+        self.session = session
+        self.doc = doc_maker.create_doc(
+            api, self.HYDRUS_SERVER_URL, self.API_NAME)
+        test_classes = doc_parse.get_classes(api)
+        self.test_class = test_classes[0]["title"]
```

Comment:
You should use random function to get the class title from list. That'll make tests more generalized.

Reply:
@vaibhavchellani ^^ We need to merge this.

Reply:
sorry totally missed this 
updating

Reply:
@xadahiya the test are failing in some cases where we randomly get collections instead of class , the tests are made for classes only i think , because if its a collection it wont pass the condition in gen_dummy_object

Reply:
@vaibhavchellani fix this like we discussed yesterday, then we can merge this.

Reply:
@vaibhavchellani Please fix this, we need to merge.

Reply:
@vaibhavchellani Please fix this, we need to merge.